### PR TITLE
[app_dart] Remove Cocoon GitHub models

### PR DESCRIPTION
**Pull request by CaseyHillers**
* Migrate to relying on the `PullRequest` instead of splitting it into pieces (like `commitSha`, `slug`, `prNumber`)
  * This makes several function signatures cleaner
  * Enables future work of relying on other pull request data throughout the code
* Remove the Cocoon github `CheckSuiteEvent` and `CheckRunEvent`
* Removed `commitSha` query param from reset-try-task as it can be pulled from the pr number
* Upgrade package:github to 8.2.1

This is follow up from https://github.com/SpinlockLabs/github.dart/pull/269

Simplifies changes necessary for https://github.com/flutter/cocoon/pull/1389 (pass the PullRequest object instead of piece meal parts we have to plumb everywhere)

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [X] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] All existing and new tests are passing.

### CODE REVIEW

File: app_dart/lib/src/service/luci_build_service.dart [hunk 1]
```diff
@@ -411,19 +391,18 @@ class LuciBuildService {
...
 
-  /// Sends a [BuildBucket.scheduleBuild] request using [CheckRunEvent]. It
-  /// returns [true] if it is able to send the scheduleBuildRequest or [false]
-  /// if not.
-  Future<bool> rescheduleUsingCheckRunEvent(CheckRunEvent checkRunEvent) async {
+  /// Sends [ScheduleBuildRequest] for [pullRequest] using [checkRunEvent].
+  /// 
+  /// Returns true if it is able to send the scheduleBuildRequest. Otherwise, false.
+  Future<bool> rescheduleUsingCheckRunEvent(github.PullRequest pullRequest, CheckRunEvent checkRunEvent) async {
     final github.RepositorySlug slug = checkRunEvent.repository!.slug();
```
Comment (keyonghan) [inline app_dart/lib/src/service/luci_build_service.dart:398]: remove?

File: app_dart/lib/src/service/luci_build_service.dart [hunk 2]
```diff
@@ -411,19 +391,18 @@ class LuciBuildService {
...
     final Map<String, dynamic> userData = <String, dynamic>{};
     final String? commitSha = checkRunEvent.checkRun!.headSha;
```
Comment (keyonghan) [inline app_dart/lib/src/service/luci_build_service.dart:400]: ditto.

File: app_dart/lib/src/service/luci_build_service.dart [hunk 3]
```diff
@@ -411,19 +391,18 @@ class LuciBuildService {
...
     final String? builderName = checkRunEvent.checkRun!.name;
     final github.CheckRun githubCheckRun = await githubChecksUtil.createCheckRun(
       config,
-      slug,
+      pullRequest,
       checkRunEvent.checkRun!.name,
-      commitSha,
     );
     final Iterable<Build> builds = await getTryBuilds(slug, commitSha, builderName);
```
Comment (keyonghan) [inline app_dart/lib/src/service/luci_build_service.dart:407]: update `slug`, `commitSha`?

File: app_dart/test/model/github/checks_test_data.dart [hunk 4]
```diff
@@ -15,7 +15,7 @@ String checkSuiteTemplate(String action) => '''\
         "head_branch": "update_licenses",
         "head_sha": "dabc07b74c555c9952f7b63e139f2bb83b75250f",
         "status": "queued",
-        "conclusion": null,
+        "conclusion": "success",
```
Comment (keyonghan) [inline app_dart/test/model/github/checks_test_data.dart:18]: Is this file still needed?

File: app_dart/lib/src/service/luci_build_service.dart (hunk 2 above)
Comment (CaseyHillers) [reply app_dart/lib/src/service/luci_build_service.dart:399]: I think we want to rely on the check run for consistency. There's a chance a user may rerun a check + push a new commit, which case we could result in 2 runs of the latest sha.

File: app_dart/lib/src/service/luci_build_service.dart (hunk 1 above)
Comment (CaseyHillers) [reply app_dart/lib/src/service/luci_build_service.dart]: Swapped to rely on the `pullRequest`, and i'm going to leave the shorthand here as it's used several places in this method
